src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -487,7 +489,7 @@ private SearchResult CreateSearchResult(SearchOptions searchOptions, IEnumerable
                             queryRequestOptions,
                             includeResponse.continuationToken,
                             false,
-                            minimumDesiredPercentage: 100, // since we are executing across partitions in parallel, we will have results from all partitions, so fill up the page.
+                            minimumDesiredPercentage: (int)Math.Ceiling(maxIncludeCount * 100.0 / queryRequestOptions.MaxItemCount.Value), // ensure we fill up to maxIncludeCount in the initial query, as the SDK does not parallelize queries when there is a continuation token
```

Comment:
Just clarifying with the description, is/was this 100 results or 100% of results?

Reply:
The percentage is with respect to `queryRequestOptions.MaxItemCount`, which in this case is 10. The actual number of resources we want 100, so we are asking for 1000% or 10x `MaxItemCount`. `MaxItemCount` controls the page size requested from each physical partition.

Reply:
The percentage is with respect to 

> @johnstairs can you also create a branch off main with any changes from this hotfix branch we need. I think is going to be cleaner given its age

@brendankowitz, yes, I will also port this to main (the implementation will be a bit different)